src/ApplicationInsights.Kubernetes/K8sEnvironmentFactory.cs

```diff
@@ -42,15 +43,28 @@ internal class K8sEnvironmentFactory : IK8sEnvironmentFactory
                 using (IKubeHttpClient httpClient = _httpClientFactory.Create(_httpClientSettings))
                 using (K8sQueryClient queryClient = _k8sQueryClientFactory.Create(httpClient))
                 {
-                    string containerId = _httpClientSettings.ContainerId;
+                    // TODO: See if there's better way to fetch the container id
+                    K8sPod myPod = await SpinWaitUntilGetPod(timeoutAt, queryClient).ConfigureAwait(false);
+                    string containerId = null;
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        // For Windows, is there a way to fetch the current container id from within the container?
+                        containerId = myPod.Status.ContainerStatuses.First().ContainerID;
+                    }
+                    else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        // For Linux, container id could be fetched directly from cGroup.
+                        containerId = _httpClientSettings.ContainerId;
+                    }
+                    // ~
+
                     if (await SpinWaitContainerReady(timeoutAt, queryClient, containerId).ConfigureAwait(false))
```

Comment:
Is this still necessary given that you are waiting for the pod to be ready on line 47?

Reply:
They look similar but there is a minor difference. Line 47 is wait until the pod info is query-able. That is the earliest point where container id could be fetched; Here is waiting until the status of the pod become 'Ready'. There is a small gap between the 2 status.

Reply:
makes sense, thx for clarification